File: packages/public-domain/src/types/Action.ts
export type LiteralPatternItem = { type: 'literal'; value: string };
export type VariablePatternItem = { type: 'variable'; name: string };

export type PatternItem = LiteralPatternItem | VariablePatternItem;

export interface ActionCommand {
  pattern: PatternItem[];
  variables: string[];
}

export type ActionDefinition = {
  id: string;
  affinityId: string;
  name: string;
  command: ActionCommand;
  version: number;
};

export type Action = {
  id: string;
  agentId: string;
  variableMapping: VariableMapping;
};

export type VariableMapping = { [variableName: string]: string };

File: grafana-plugin/pkg/datasource/domain/action.go
package domain

type LiteralPatternItem struct {
	Value string `json:"value"`
	Type  string `json:"type"`
}

type VariablePatternItem struct {
	Name string `json:"name"`
	Type string `json:"type"`
}

type ActionCommand struct {
	Pattern   []any    `json:"pattern"`
	Variables []string `json:"variables"`
}

type ActionDefinition struct {
	Id         string        `json:"id"`
	AffinityId AffinityId    `json:"affinityId"`
	Name       string        `json:"name"`
	Command    ActionCommand `json:"command"`
	Version    int           `json:"version"`
}
